app/controllers/check_answers_controller.rb

```diff
@@ -10,16 +10,16 @@ def update
       if Steps::Helper.last_step_in_group?(session_data, step)
         next_step = next_check_answer_step(step)
         if next_step
-          redirect_to wizard_path next_step
+          redirect_to helpers.check_step_path_from_step(next_step, assessment_code)
```

Comment:
Oh that's interesting. So I see that there's a suggestion this may be bad practice in [this article](https://www.rubyguides.com/2020/01/rails-helpers/) but it doesn't elaborate as to why. I can think the criticisms might be:

(a) namespace issues. If you have two helper methods in different modules with the same name, it's not clear which you'll get from `helpers.helper_name`. But that's really an issue with Rails helpers in general, and one which I think we all know to work around.

(b) doing too much in controllers. Broadly, controllers should be for handling redirects, renders, cookies, session data, and not much else. Too much use of helpers would perhaps be an indication of messy design, if the same business logic is being called from controllers and views etc. But in this case this is a helper that is being used _for a redirect_ so I don't feel like that applies.

I can't find any other criticisms of using `helpers` from a quick Google. If you know of any do please send them my way.

What we're doing here is creating a method that's like the ones in Rails.application.routes.url_helpers (like "root_path", "new_check_url" etc, but has one extra layer of indirection in that we provide a `step` and that gets converted automatically into the `step_url_fragment` in the URL. In the same way that other url_helpers are routinely referenced in both controllers and views, this method is needed in both. We _could_ have wrapped it up into a service object, but because it's basically a one-liner, that felt a little bit overly heavy, so for me a helper method felt appropriate.

Reply:
Thanks for explaining! I just wasn't super familiar with the use of `helpers` in controllers because it was a pattern I hadn't seen much before, so your comment is really useful. Rationale makes sense to me! I also read this article briefly but it doesn't really elaborate that much on the use of helpers in controllers either https://blog.appsignal.com/2023/02/01/a-guide-to-rails-view-helpers.html